### DISCUSSION TITLE
2006/04/08/%E7%94%A8-labview-%E7%BC%96%E5%86%99-wizard-%E7%B1%BB%E5%9E%8B%E7%9A%84%E5%BA%94%E7%94%A8%E7%A8%8B%E5%BA%8F

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#177] [Announcements]
# 2006/04/08/%E7%94%A8-labview-%E7%BC%96%E5%86%99-wizard-%E7%B1%BB%E5%9E%8B%E7%9A%84%E5%BA%94%E7%94%A8%E7%A8%8B%E5%BA%8F

Wizard 向导类型的程序，指的是可以类似安装程序那样，一步一步地指导用户完成功能的应用程序。这类程序极为常见。它一边要求用户提供必要的信息，一边给用户发出指导性的意见和反馈。这样，即便是新用户也可以轻松完成任务。但是，向导类型的程序虽然方便了用户，却是增加了开发人员工作的复杂度。

https://blog.qizhen.xyz/2006/04/08/%E7%94%A8-labview-%E7%BC%96%E5%86%99-wizard-%E7%B1%BB%E5%9E%8B%E7%9A%84%E5%BA%94%E7%94%A8%E7%A8%8B%E5%BA%8F?_highlight=wizar

**Comment by ruanqizhen** [comment]
之前博客下的留言：

Qizhen
2010年01月15日 13:04 编辑
4 打不开是因为我写到这里时思路发生了变化。后来就没有再按照原定的计划写了。第四部分打算写的是XControl+OOP。我后来的其他一些文章对这两部分有简略的介绍，书里写的更详细一些其他。

回复
枫桦
2010年01月14日 19:43 编辑
很期待阮先生的“用 LabVIEW 编写 Wizard 类型的应用程序 4”，可惜博客上打不开，不知道阮先生的新书中是否包含这部分类容？

回复
Qizhen
2007年08月30日 11:49 编辑
这个文章下一篇就是写如何利用 LVOOP 改进向导程序的架构，但是还一直没时间开始写呢 ^_^*

回复
竹贤
2007年08月30日 08:45 编辑
 你的文中说：“现在的程序结构还是有些令我不太满意的地方-重复的代码太多。”“直到 LabVIEW 支持了面向对象的编程之后，我们才终于找到了一个完美的解决这一问题的方案。”完美方案是指现在你写的方案吗？

回复
清扬
2007年08月24日 22:39 编辑
个人感觉使用subpanel调用起来速度有点慢,而且画面很容易出现错位.
我一开始也想这样做,后来放弃了,最后改用TAB CONTROL做了,所有的画面都在不同的页面上做,程序里在不同的tab嵌入不同的子程序(子vi不显示其前面板). 感觉这样速度快多了,而且在不同的步骤间传递数据起来也很方便.
很想知道你这个程序的反应速度怎么样,还有就是wizard在不同页面之间如果需要传递一些数据是怎么实现的呢?最后就是使用面向对象的编程方法,很想学习下.希望能把代码开放下,让我学习学习.~
期待ing…..

回复
